### ISSUE TITLE
--disable deep_nesting has no effect (the rule is not a registered pattern)

### ISSUE THREAD

**Issue by CognitiveDecisions** [rsionnach/sloppylint#24]
`deep_nesting` is emitted directly by the AST visitor:

https://github.com/rsionnach/sloppylint/blob/e0e3ced/src/sloppy/analyzers/ast_analyzer.py#L103

```python
if self.nesting_depth > 4:
    self.issues.append(Issue(pattern_id="deep_nesting", ...))
```

but `--disable` only filters registered patterns:

https://github.com/rsionnach/sloppylint/blob/e0e3ced/src/sloppy/detector.py#L42

```python
self.patterns = [p for p in get_all_patterns() if p.id not in self.disabled_patterns]
```

`deep_nesting` is not in `get_all_patterns()`, so it cannot be turned off by any flag or config key.

### Repro, with a control

```
$ sloppylint n.py --disable deep_nesting --format compact | grep -c deep_nesting
1                                    # still reported

$ sloppylint m.py --disable hallucinated_method --format compact | grep -c hallucinated_method
0                                    # control: a registered pattern is suppressed
```

The control matters, since it isolates this to the rule rather than to the flag.

### Effect

On a scan of 11 repositories this rule was 29% of total output (1,308 of 4,519 findings) with no way to silence it.

### Suggested fix

Make it a real `ASTPattern` with `id = "deep_nesting"` and register it in `STRUCTURE_PATTERNS`, so it goes through the same filter as everything else. The nesting depth would need to live on the pattern or be passed through `check_node`, which is the only real work involved.

Related: #25 (the same rule counts `elif` as nesting).

Found on 0.5.1 (`e0e3ced`).
